exammode/models.py

```diff
@@ -0,0 +1,107 @@
+from django.db import models
+from django.urls import reverse
+from django.utils import timezone
+
+from course.models import CourseInstance, CourseModule
+from exercise.exercise_models import LearningObject
+from userprofile.models import UserProfile
+
+# Create your models here.
+
+
+class ActiveExamSessionManager(models.Manager):
+
+    def get_queryset(self):
+        initial_queryset = super().get_queryset()
+        queryset = [q for q in initial_queryset if (
```

Comment:
Why do you loop over this queryset? You should use queryset methods such as `filter` if you want to filter the queryset.

Reply:
In addtion, this method returns a list, but it should return a `QuerySet`. This is also solved by using the QuerySet filter method.

Reply:
https://docs.djangoproject.com/en/2.2/topics/db/managers/#modifying-a-manager-s-initial-queryset

Reply:
It seems that 'get_queryset' method is never called.

Reply:
You know that `get_queryset` is part of the Django API and Django could call that method as part of some API. So it is normal that the method is not directly called in the A+ code.

https://docs.djangoproject.com/en/2.2/topics/db/managers/#modifying-a-manager-s-initial-queryset

Reply:
The `get_queryset` method is modified to be like the previous `is_active` method, which is now removed. New version uses Django's `filter()` method.